src/board/devices/output/7seg.ts

```diff
@@ -22,6 +43,56 @@ export class SEVENseg extends Device {
     }
     return this.findSegByte(position).isBitSet(position % 8)
   }
+  public getDisplay(display: number): boolean[]{
+    if( this.isBin()){
+     return this.getBinDisplay(display)
+    }else{
+
+    }
+    return [true,false,true,false,true,true,true,true]
+  }
+  /**
+   * Returns true if seg with given position is on.
+   *
+   * @returns true if segment is turned on
+   */
+  private isBin():boolean{
+    return (this.memory.readWord(this.startAddressBin)!=this.oldBin)||(this.memory.readWord(this.endAddressBin)!=this.oldBin)
+  }
+
+  private toSevensegNr(inByte: Byte):boolean[]{
+    return []
+  }
+  private getBinDisplay(display: number): boolean[]{
+    let arr: boolean[] = []
+    switch ( display ) {
+      case 0:
+        arr = this.toSevensegNr(this.memory.readByte(this.startAddressBin))
+        break;
+      case 1:
+        this.memory.readByte(this.startAddressBin)
+        break;
+      case 2:
+        this.memory.readByte(this.endAddressBin)
+        break;
+      case 3:
+        this.memory.readByte(this.endAddressBin)
```

Comment:
as in case 0 push values to arr and use toSevensegNr method (which has to be implemented)